tests/storage_and_cookie_security_tests.rs

```diff
@@ -0,0 +1,394 @@
+        // Wait for both tasks to complete
+        let (result1, result2) = tokio::try_join!(handle1, handle2).unwrap();
+        
+        // Verify both accounts were stored
+        assert!(storage.query_account_by_user_id("user1@example.com").await.unwrap().is_some());
+        assert!(storage.query_account_by_user_id("user2@example.com").await.unwrap().is_some());
+    }
+
+    #[tokio::test]
+    async fn test_secret_storage_security() {
+        let storage = Arc::new(MemorySecretStorage::default());
+        let account_id = Uuid::now_v7();
+        let password = "sensitive_password";
+
+        let secret = Secret::new(&account_id, password, Argon2Hasher).unwrap();
+        storage.store_secret(secret).await.unwrap();
+
+        // Verify correct password works
+        let correct_creds = Credentials::new(&account_id, password);
+        assert_eq!(
+            storage.verify_credentials(correct_creds).await.unwrap(),
+            VerificationResult::Ok
+        );
+
+        // Verify wrong password fails
+        let wrong_creds = Credentials::new(&account_id, "wrong_password");
+        assert_eq!(
+            storage.verify_credentials(wrong_creds).await.unwrap(),
+            VerificationResult::Unauthorized
+        );
+    }
+
+    #[tokio::test]
+    async fn test_storage_with_duplicate_user_ids() {
+        let storage = Arc::new(MemoryAccountStorage::<Role, Group>::default());
+        let user_id = "duplicate@example.com";
+
+        let account1 = Account::new(user_id, &[Role::User], &[Group::new("group1")]);
+        let account2 = Account::new(user_id, &[Role::Admin], &[Group::new("group2")]);
+
+        // Store first account
+        let _stored1 = storage.store_account(account1).await.unwrap();
+        
+        // Store second account with same user_id
+        let _stored2 = storage.store_account(account2).await.unwrap();
+
+        // Query should return one of them (implementation dependent)
+        let queried = storage.query_account_by_user_id(user_id).await.unwrap();
+        assert!(queried.is_some());
+    }
+
+    #[tokio::test]
+    async fn test_account_deletion_security() {
+        let account_storage = Arc::new(MemoryAccountStorage::<Role, Group>::default());
+        let secret_storage = Arc::new(MemorySecretStorage::default());
+
+        // Create an account with secret
+        let account = AccountInsertService::insert("user@example.com", "password")
+            .with_roles(vec![Role::User])
+            .into_storages(Arc::clone(&account_storage), Arc::clone(&secret_storage))
+            .await
+            .unwrap()
+            .unwrap();
+
+        let user_id = &account.user_id;
+
+        // Verify account exists
+        assert!(account_storage.query_account_by_user_id(user_id).await.unwrap().is_some());
+
+        // Delete the account by user_id
+        account_storage.delete_account(user_id).await.unwrap();
+        secret_storage.delete_secret(&account.account_id).await.unwrap();
```

Comment:
There is also the `AccountDeletionService` that handles deletion of account and secrets. Please use it instead.